todo-list/src/todo.ts

```diff
@@ -0,0 +1,65 @@
+interface Todo {
+    id: number;
+    text: string;
+    completed: boolean;
+  }
+  
+  class TodoList {
+    todos: Todo[] = [];
+  
+    addTodo(text: string): void {
+      const newTodo: Todo = {
+        id: Date.now(),
+        text,
+        completed: false,
+      };
+      this.todos.push(newTodo);
+      this.render();
+    }
+  
+    deleteTodo(id: number): void {
+      this.todos = this.todos.filter(todo => todo.id !== id);
+      this.render();
+    }
+  
+    toggleComplete(id: number): void {
+      const todo = this.todos.find(todo => todo.id === id);
+      if (todo) {
+        todo.completed = !todo.completed;
+      }
+      this.render();
+    }
+  
+    render(): void {
+      const todoListElement = document.getElementById('todo-list')!;
+      todoListElement.innerHTML = '';
+  
+      this.todos.forEach(todo => {
+        const li = document.createElement('li');
+        li.className = todo.completed ? 'completed' : '';
+        li.innerHTML = `
+          ${todo.text}
```

Comment:
HINT: Be careful of this, this is vulnerable to XSS
https://owasp.org/www-community/attacks/xss/